Feat/#84 polish company create registration flow

TerrellLinnell opened the pull request.

Commit:
Merge pull request #3 from codexstanford/develop

feat(BingNewsAPI): adds dynamic news integration for company pages.

Commit:
Merge pull request #4 from codexstanford/develop

fix(build): rollback seo qraphql query

Commit:
Merge pull request #6 from codexstanford/develop

Fix Bugs & Implement Better Auth/Session Support

Commit:
Merge branch 'develop'

Commit:
hotfix: tweak gatsby-node query structure

TerrellLinnell committed:
fix: point frontend to local backend

make necessary changes to point the frontend to use a local docker backend

#NO_ISSUE_ID

TerrellLinnell committed:
feat: Allow user to exit create company modal

Allow user to exit create company modal, utilizes MUI onBackdropClick

#65

TerrellLinnell committed:
feat: Allow user to exit create company modal

Allow user to exit create company modal, utilizes MUI onBackdropClick

#65

TerrellLinnell committed:
feat: Allow user to exit create company modal

Allow user to exit create company modal, utilizes MUI onBackdropClick

#65

TerrellLinnell committed:
feat: Allow user to exit create company modal

Allow user to exit create company modal, utilizes MUI onBackdropClick

#65

TerrellLinnell committed:
feat: Allow user to exit create company modal

Allow user to exit create company modal, utilizes MUI onBackdropClick

#65

TerrellLinnell committed:
Merge pull request #88 from codexstanford/feat/allow-users-to-click-out-of-company-create-modal

Feat/allow users to click out of company create modal

TerrellLinnell committed:
feat: Make get listed button functional

make get listed button take you to login if you are not authenticated or if you are then bring up create company dialog

#77

TerrellLinnell committed:
feat: Make get listed button functional

make get listed button take you to login if you are not authenticated or if you are then bring up create company dialog

#77

TerrellLinnell committed:
feat: Make get listed button functional

make get listed button take you to login if you are not authenticated or if you are then bring up create company dialog

#77

TerrellLinnell committed:
feat: Make get listed button functional

make get listed button take you to login if you are not authenticated or if you are then bring up create company dialog

#77

TerrellLinnell committed:
Fix: Remove console logs

TerrellLinnell committed:
Fix: Remove console logs

TerrellLinnell committed:
Fix: Remove console logs

TerrellLinnell committed:
Fix: Remove console logs

TerrellLinnell committed:
Merge pull request #89 from codexstanford/feat/make-get-listed-button-in-navigation-functional

Feat/make get listed button in navigation functional

TerrellLinnell committed:
feat: add validation to basic section of create company

Add validation to basic section of create company form using formik and yup object schema validation

#67

TerrellLinnell committed:
feat: add validation to basic section of create company

Add validation to basic section of create company form using formik and yup object schema validation

#67

TerrellLinnell committed:
feat: add validation to basic section of create company

Add validation to basic section of create company form using formik and yup object schema validation

#67

TerrellLinnell committed:
feat: add validation to basic section of create company

Add validation to basic section of create company form using formik and yup object schema validation

#67

TerrellLinnell committed:
Merge pull request #90 from codexstanford/feat/add-field-level-validation-to-basics-section

feat: add validation to basic section of create company

TerrellLinnell committed:
feat: form level validation for create company

form validation works for form items that have validation, biolerplate in place for when rest of form gets validation

#72

TerrellLinnell committed:
feat: form level validation for create company

form validation works for form items that have validation, biolerplate in place for when rest of form gets validation

#72

TerrellLinnell committed:
feat: form level validation for create company

form validation works for form items that have validation, biolerplate in place for when rest of form gets validation

#72

TerrellLinnell committed:
feat: form level validation for create company

form level validation partially implemented

#72

TerrellLinnell committed:
feat: form level validation for create company

form level validation partially implemented

#72

TerrellLinnell committed:
feat: form level validation for create company

form level validation partially implemented

#72

TerrellLinnell committed:
feat: form level validation for create company

form level validation partially implemented

#72

Commit:
chore: support local dev without code edits

Commit:
chore: support local dev without code edits

TerrellLinnell committed:
feat: Add form level validation to company create

Added form level validation to company create form, custom validation check for errors and touched fields, error message outside sections pointing to correct section

#72

TerrellLinnell committed:
feat: Add form level validation to company create

Added form level validation to company create form, custom validation check for errors and touched fields, error message outside sections pointing to correct section

#72

TerrellLinnell committed:
feat: Add form level validation to company create

Added form level validation to company create form, custom validation check for errors and touched fields, error message outside sections pointing to correct section

#72

TerrellLinnell committed:
feat: Add form level validation to company create

Added form level validation to company create form, custom validation check for errors and touched fields, error message outside sections pointing to correct section

#72

TerrellLinnell committed:
Merge pull request #93 from codexstanford/feat/form-level-validation-create-company

Feat/form level validation create company

TerrellLinnell committed:
chore: Merge remote branch into local

TerrellLinnell committed:
chore: Pretty prinit company categories

Create formatCatagory function to format company categories going in and out

#66

TerrellLinnell committed:
chore: Pretty prinit company categories

Create formatCatagory function to format company categories going in and out

#66

TerrellLinnell committed:
chore: Pretty prinit company categories

Create formatCatagory function to format company categories going in and out

#66

TerrellLinnell committed:
Merge pull request #94 from codexstanford/chore/pretty-print-company-categories

chore: Pretty prinit company categories

TerrellLinnell committed:
chore: Restyle company location suggestion box

created new file to avoid cobining classes and mui theme errors, moved paper to new file, styled correctly in both and all future company create modals

#69

TerrellLinnell committed:
chore: Restyle company location suggestion box

created new file to avoid cobining classes and mui theme errors, moved paper to new file, styled correctly in both and all future company create modals

#69

TerrellLinnell committed:
chore: Restyle company location suggestion box

created new file to avoid cobining classes and mui theme errors, moved paper to new file, styled correctly in both and all future company create modals

#69

TerrellLinnell committed:
Merge pull request #95 from codexstanford/chore/restyle-company-location-suggestion

chore: Restyle company location suggestion box

TerrellLinnell committed:
chore: Restyle logo input

Restyled logo input to be smaller and responsive

#48

TerrellLinnell committed:
chore: Restyle logo input

Restyled logo input to be smaller and responsive

#48

TerrellLinnell committed:
chore: Restyle logo input

Restyled logo input to be smaller and responsive

#48

TerrellLinnell committed:
Merge pull request #96 from codexstanford/chore/restyle-logo-input

 chore: Restyle logo input

TerrellLinnell committed:
chore: Restyle company categories in create company form

Moved chips below input, made categories section height responsive to inside content

#71

TerrellLinnell committed:
chore: Restyle company categories in create company form

Moved chips below input, made categories section height responsive to inside content

#71

TerrellLinnell committed:
chore: Restyle company categories in create company form

Moved chips below input, made categories section height responsive to inside content

#71

TerrellLinnell committed:
Merge pull request #99 from codexstanford/chore/restyle-company-categories-input

chore: Restyle company categories in create company form

TerrellLinnell committed:
fix: Fix typo in form level errors

Fixed typo in form level errors for create company form

#72

TerrellLinnell committed:
fix: Fix typo in form level errors

Fixed typo in form level errors for create company form

#72

TerrellLinnell committed:
fix: Fix typo in form level errors

Fixed typo in form level errors for create company form

#72

TerrellLinnell committed:
Merge pull request #100 from codexstanford/fix/fix-typo-in-create-company-form-validation

fix: Fix typo in form level errors

TerrellLinnell committed:
fix: Restyle logo in create company

Restyled logo input, images are now confined to the space available

#48

TerrellLinnell committed:
fix: Restyle logo in create company

Restyled logo input, images are now confined to the space available

#48

TerrellLinnell committed:
fix: Restyle logo in create company

Restyled logo input, images are now confined to the space available

#48

TerrellLinnell committed:
Merge pull request #101 from codexstanford/fix/logo-input-create-company

fix: Restyle logo in create company

Commit:
feat: set current user to admin on company create

closes techlist-frontend-web/#78

Commit:
feat: set current user to admin on company create

closes techlist-frontend-web/#78

Commit:
feat: set current user to admin on company create

closes techlist-frontend-web/#78

Commit:
feat: set current user to admin on company create

closes techlist-frontend-web/#78

TerrellLinnell committed:
Merge pull request #102 from codexstanford/feat/auth

feat: set current user to admin on company create

Commit:
Merge branch 'feat/#84-polish-company-create-registration-flow' of github.com:codexstanford/techlist-frontend-web into feat/#84-polish-company-create-registration-flow

TerrellLinnell committed:
chore: Make create company form mobile friendly

Made all sections responsive to screen size changes, this includes a change up in the preview window, fixed form validation issue where touching some fields did not set them as touched fields, categories had to use onFocus instead of onBlur, reworked form level error to look better with long error messages

#73

TerrellLinnell committed:
chore: Make create company form mobile friendly

Made all sections responsive to screen size changes, this includes a change up in the preview window, fixed form validation issue where touching some fields did not set them as touched fields, categories had to use onFocus instead of onBlur, reworked form level error to look better with long error messages

#73

TerrellLinnell committed:
fix: Check for no errors before displaying

form errored out due to having no errors and trying to read properties off of null

#73

TerrellLinnell committed:
fix: Check for no errors before displaying

form errored out due to having no errors and trying to read properties off of null

#73

TerrellLinnell committed:
fix: Remove StylesProvider from files

remove StylesProvider it is uneccessary because we have a custom insertion point

#73

TerrellLinnell committed:
fix: Remove StylesProvider from files

remove StylesProvider it is uneccessary because we have a custom insertion point

#73

TerrellLinnell committed:
Merge pull request #103 from codexstanford/chore/make-company-create-profile-mobile-friendly

chore: Make create company form mobile friendly

TerrellLinnell committed:
fix: Resize logo in preview

Logo was getting pushed around, stretched, and squashed. Gave it min/max height/width now it is consistent

#73

TerrellLinnell committed:
fix: Resize logo in preview

Logo was getting pushed around, stretched, and squashed. Gave it min/max height/width now it is consistent

#73

TerrellLinnell committed:
Merge pull request #104 from codexstanford/fix/logo-size-preview

fix: Resize logo in preview

TerrellLinnell committed:
fix: Close mobile nav when dialog opens

mobile nav was swiping when trying to scroll the create company dialog, now when the dialog opens the nav closes preventing it from being swiped

TerrellLinnell committed:
fix: Close mobile nav when dialog opens

mobile nav was swiping when trying to scroll the create company dialog, now when the dialog opens the nav closes preventing it from being swiped

TerrellLinnell committed:
Merge pull request #106 from codexstanford/fix/mobile-nav-swipes-on-dialog-scroll-mobile

Fix/mobile nav swipes on dialog scroll mobile

TerrellLinnell committed:
chore: Make company create dialog full screen

TerrellLinnell committed:
chore: Make company create dialog full screen

TerrellLinnell committed:
Merge pull request #107 from codexstanford/chore/make-company-create-full-screen-on-mobile

Chore/make company create full screen on mobile

TerrellLinnell committed:
fix: Make create company mutation work

Currently does not work but we have users now and the mutation is firing

medelman17 commented on src/apps/admin/components/codex.textinput.js:
```diff
@@ -24,9 +25,10 @@ export default function CodeXTextField({
         fullWidth={false}
         InputLabelProps={{}}
         value={value}
+        onBlur={onBlur}
```
Was this not already being passed down via `...rest`? Is there a reason that it needs to be explicitly stated?

medelman17 commented on src/apps/admin/components/select/helpers/renderSuggestions.js:
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
-import MenuItem from '@material-ui/core/MenuItem';
 import PropTypes from 'prop-types';
+import MenuItem from '@material-ui/core/MenuItem';
```
This seems unnecessary.

TerrellLinnell replied:
What do you mean?
MenuItem is used in the component and that's how the material ui docs say to import it.

medelman17 replied:
It's fine, but it looked like it was already imported and just the line number changed. Don't worry about it.

medelman17 commented on src/apps/admin/features/company.create/components/basics.js:
```diff
@@ -30,7 +30,7 @@ export function Basics({ errors, touched, classes, targetMarkets, ...rest }) {
         label="Description"
       />
-      <div style={{ display: 'flex' }}>
+      <YearAndMarketLayout>
```
Make this name either more generic or more descriptive. From the styling below, it looks like we would probably use this many places? So, I'm thinking more generic would be appropriate.

TerrellLinnell replied:
changed to FlexLayoutMobile

medelman17 commented on src/apps/admin/features/company.create/components/categories/helpers/formatCategory.js:
```diff
@@ -0,0 +1,18 @@
+const formatCategory = str => {
+  if (str.toUpperCase() != str) {
```
Did you intend to use a loose-y `not equals` here or should we correct to strict inequality?

TerrellLinnell replied:
changed to strict inequality

medelman17 commented on src/apps/admin/features/company.create/components/categories/helpers/formatCategory.js:
```diff
@@ -0,0 +1,18 @@
+const formatCategory = str => {
+  if (str.toUpperCase() != str) {
+    const newStr = str.toUpperCase().replace(' ', '_');
```
Rather than declaring a new variable and assigning it the value we immediately thereafter return, let's just return the value without declaring the new variable?

TerrellLinnell replied:
correctamundo

medelman17 commented on src/apps/admin/features/company.create/components/categories/helpers/formatCategory.js:
```diff
@@ -0,0 +1,18 @@
+        return a.toUpperCase();
+      });
+
+    return newStr;
```
Same as comment above.

medelman17 commented on src/apps/admin/features/company.create/components/categories/helpers/renderInput.js:
```diff
@@ -14,6 +16,7 @@ export function renderInput(inputProps) {
           input: classes.inputInput,
         },
         ...InputProps,
+        onFocus: onBlur, // won't run passsing onBlur have to use onFocus
```
Huh? This can't be right.

TerrellLinnell replied:
I passed it a onBlur prop every way I could think of but it really just will not take it, the only way I could get it to say it was touched when a user clicked on it was to set the function I was going to fire onBlur to fire on focus instead.

medelman17 commented on src/apps/admin/features/company.create/components/categories/helpers/renderSuggestions.js:
```diff
@@ -24,7 +25,7 @@ export function renderSuggestion(suggestionProps) {
       }}
     >
-      {suggestion.label}
+      {formatCategory(suggestion.label)}
```
We should probably either include a null check here or in the `formatCategory` function.

TerrellLinnell replied:
null check now in renderSuggestions file

medelman17 commented on src/apps/admin/features/company.create/components/categories/index.js:
```diff
@@ -75,30 +76,21 @@ export function DownshiftMultiple(props) {
             })}
 
-            {isOpen ? (
+            {true ? (
```
Huh?

TerrellLinnell replied:
While messing with it I had to leave it open in order to inspect it to see what was happening with the styles, I forgot to change it back

medelman17 commented on src/apps/admin/features/company.create/components/links/links.input.js:
```diff
@@ -80,6 +81,27 @@ function CompanyLinksInput(props) {
+`;
+
+const ButtonsContainer = styled.div`
+  display: inline-block;
```
This is being immediately overwritten below, no?

TerrellLinnell replied:
yep

medelman17 commented on src/apps/admin/features/company.create/components/location/locationPaper.js:
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import Paper from '@material-ui/core/Paper';
+import withStyles from '@material-ui/core/styles/withStyles';
+import { styles } from '../../../../config/styles';
```
Please use the `useStyles` api from MaterialUI to implement the styling for this component. It's too far away from this file that it's almost certain that we'll forget it's coupled. 

https://material-ui.com/styles/basics/

TerrellLinnell replied:
Use styles doesn't work without a theme provider, I'm unsure if you want a theme provider in the app so I opted for a similar approach except styles is defined within the file.

medelman17 replied:
It should work. But make sure you're importing it like so: 

```import { makeStyles } from '@material-ui/styles';```

Then you should be able to just return an object: 

```
const useStyles = makeStyles({
  ...
})
```

TerrellLinnell replied:
The problem is the theme object is null when using it like that  without a theme provider and in order to style the paper like the other papers we have in the body I need the theme

medelman17 commented on src/apps/admin/features/company.create/components/logo.js:
```diff
@@ -32,40 +35,54 @@ export function Logo({
 }
 
+const InputContainer = styled.div`
+  min-width: 99%;
```
What are you trying to do here? There has to be a 'more refined' way, no?

TerrellLinnell replied:
replaced with width: 100%

medelman17 commented on src/apps/admin/features/company.create/components/logo.js:
```diff
@@ -32,40 +35,54 @@ export function Logo({
+  max-width: 100%;
+`;
+
+const InputWrapper = styled(InputContainer)`
```
Not sure why we need to re-style this if all it's doing is adding `display: flex`, etc.

TerrellLinnell replied:
I was actually able to remove this component all together

medelman17 commented on src/apps/admin/features/company.create/components/preview.js:
```diff
@@ -11,17 +11,23 @@ const useStyles = makeStyles(theme => ({
   cover: {
-    width: 151,
-    height: 151,
+    minWidth: 151,
```
(1) let's move away from hardcoded values, instead taking advantage of relative units and flex-box; 
(2) there has to be a more elegant solution;
(3) if we set max and min to be the same value, aren't we just doing the same thing as setting the actual property value as we were before?

TerrellLinnell replied:
The logo to look right needs to stay that size at all times though

medelman17 replied:
Ok, We'll have to revisit this, but I don't want to gum things up now.

medelman17 commented on src/apps/admin/features/company.create/helpers/renderExpansionPanel.js:
```diff
@@ -14,10 +14,14 @@ export function CodeXExpansionPanel({
   titleComponent,
   titleStyle,
   titleVariant,
+  error,
```
I don't think this component needs to know whether an error has occurred or not. Let's just pass down the appropriate styling from above?

TerrellLinnell replied:
Not sure I agree, I see it with the border we can style that but if we want to do something fancier, like you said maybe an exclamation point by the name, we're going to have to handle that sort of addition within the component itself. By passing an error state it sort of future proofs it so we can make that error state be whatever we want and we only have to implement a custom error state once in the expansion component. Then it will be extended to all expansions that utilize form validation if we put form validation into the rest of the form.

medelman17 commented on src/apps/admin/features/company.create/index.js:
```diff
@@ -39,39 +42,130 @@ export function CreateCompany({ handleCompanyCreate, classes, ...props }) {
           ...rest,
         });
+
+        const intersection = Object.keys(touched).filter(element =>
```
All of these helper functions should be extracted.

medelman17 commented on src/apps/admin/features/company.create/index.js:
```diff
@@ -39,39 +42,130 @@ export function CreateCompany({ handleCompanyCreate, classes, ...props }) {
               <CodeXFormHeader text={`Create Company Profile`} />
               <Preview values={values} />
-              <CodeXExpansionPanel title="Basics">
+              {getDisplayedErrorMessage() !== null && (
```
This should be a separate component imported and rendered like `Preview` above.

medelman17 commented on src/apps/admin/features/company.create/index.js:
```diff
@@ -39,39 +42,130 @@ export function CreateCompany({ handleCompanyCreate, classes, ...props }) {
+
+              <CodeXExpansionPanel
+                title="Basics"
+                error={
```
Nope.

medelman17 commented on src/apps/admin/features/company.create/index.js:
```diff
@@ -39,39 +42,130 @@ export function CreateCompany({ handleCompanyCreate, classes, ...props }) {
-              <CodeXExpansionPanel title="Logo">
+              <CodeXExpansionPanel
+                title="Logo"
+                error={errors.logo && touched.logo ? true : false}
```
Isn't all this replicating the functionality we commented out in `CodeXTextField`? I believe I may have implemented the CodeXTextField the previous way to avoid needing to include this (very repetitive) logic higher up in the tree. 

To the extent we need more complex logic for errors, let's extract that into a separate, generic function (e.g., `getErrors`) so we don't have to gum up our render/return statement.

medelman17 commented on src/apps/admin/features/company.create/index.js:
```diff
@@ -39,39 +42,130 @@ export function CreateCompany({ handleCompanyCreate, classes, ...props }) {
+                  errors={
+                    (errors.website && touched.website) ||
+                    (errors.addresss && touched.address)
+                      ? true
```
You shouldn't need the ternary operator here if you're only looking to return a boolean.

medelman17 commented on src/apps/admin/features/company.create/index.js:
```diff
@@ -94,4 +188,28 @@ export function CreateCompany({ handleCompanyCreate, classes, ...props }) {
   );
 }
 
+const currentDate = new Date();
```
Extract all validation functionality to a separate file.

medelman17 commented on src/apps/admin/routes/company/index.js:
```diff
@@ -17,21 +18,45 @@ export default function CreateCompanyScreen({
+const StyledPaper = styled(Paper)`
+  max-width: 530px;
+  max-height: calc(100vh - 96px);
+  @media (max-width: 480px) {
```
See comment above re just setting width to `100vw` and height to `100vh` etc

medelman17 commented on src/components/header/header.three.js:
```diff
@@ -11,12 +11,15 @@ import Avatar from '@material-ui/core/Avatar';
-  console.log('DATA IN HEADER THREE:', data);
   const { user } = data;
 
+  const isUserLoggedIn = user ? true : false;
```
Why not use the same `user`-based implementation as already contained herein (below)?

medelman17 commented on src/components/header/header.three.js:
```diff
@@ -65,20 +73,35 @@ export function HeaderLeft({ sections, classes, ...props }) {
+              size="small"
+              aria-label="Get Listed"
+              component={props => (
+                <Link to="/app/login/" component={GatsbyLink} {...props} />
```
If the only difference here is whether, onClick, we're (1) showing the company create screen or (2) we're navigating the user to login, we should use the `navigate` method from `@reach/router` and just swap out that little bit of functionality rather than repeat all this code. 

It might make sense to just `handleGetListedClick` in a tiny helper function that (1) checks whether the user is authenticated and (2) acts accordingly.

TerrellLinnell replied:
If there is a user it is a normal button. If there is no user that button actually gets replaced by that link component. So they look the same but are actually 2 different components.

medelman17 replied:
Right. But isn't the entire purpose of the Link component just to navigate the user?

TerrellLinnell replied:
oh, gotcha ok

medelman17 reviewed (changes requested):
Good job. These are mostly nits. Please mark each item as resolved when addressed. Thanks!

TerrellLinnell committed:
Merge pull request #109 from codexstanford/fix/create-company-mutation

Fix/create company mutation

Commit:
fix: merge conflicts

TerrellLinnell committed:
fix: Address required fields validation, close modal on submit

target markets was required for submission, made it required, modal now closes after company creation

Commit:
fix: merge conflicts

TerrellLinnell committed:
chore: Polish company create flow

Finish last remaining fixes, implement fixes from code review

medelman17 commented:
Ok. No more time on this.

Michael Edelman
Founder & Principal
Fabulas
1650 Market Street, Suite 3600
Philadelphia, PA 19103
michael@fabulas.io<mailto:michael@fabulas.io>



From: TerrellLinnell <notifications@github.com>
Reply-To: codexstanford/techlist-frontend-web <reply@reply.github.com>
Date: Thursday, July 11, 2019 at 4:11 PM
To: codexstanford/techlist-frontend-web <techlist-frontend-web@noreply.github.com>
Cc: Michael Edelman <michael@fabulas.io>, Review requested <review_requested@noreply.github.com>
Subject: Re: [codexstanford/techlist-frontend-web] Feat/#84 polish company create registration flow (#108)


@TerrellLinnell commented on this pull request.

________________________________

In src/apps/admin/features/company.create/components/location/locationPaper.js<https://github.com/codexstanford/techlist-frontend-web/pull/108#discussion_r302722238>:

> @@ -0,0 +1,36 @@

+import React from 'react';

+import Paper from '@material-ui/core/Paper';

+import withStyles from '@material-ui/core/styles/withStyles';

+import { styles } from '../../../../config/styles';

The problem is the theme object is null when using it like that without a theme provider and in order to style the paper like the other papers we have in the body I need the theme

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub<https://github.com/codexstanford/techlist-frontend-web/pull/108?email_source=notifications&email_token=ADQ3VLNJBCTUNPDAH4PVYSDP66HYBA5CNFSM4IAC2XHKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB6GXGZQ#discussion_r302722238>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ADQ3VLOQTPOR35KDT5JEVVDP66HYBANCNFSM4IAC2XHA>.

TerrellLinnell committed:
fix: Make get listed open dialog in profile nav

Change profile nav logic from navigating to app/company to open create company dialog

TerrellLinnell committed:
Merge pull request #123 from codexstanford/fix/company-create-polishing

WIP: Fix/company create polishing
